packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/PreviousMarks/PreviousMarks.js

```diff
@@ -38,26 +35,25 @@ function PreviousMarks ({
   scale = 1
 }) {
   const {
-    /** Clone all previous marks across all frames */
-    multiImageCloneMarkers,
     /** Annotations from previous marking tasks. Each annotation is an array of marks. */
     previousAnnotations,
     /** The show/hide previous marks setting. */
     shownMarks = 'ALL'
   } = useStores(storeMapper)
 
+  /** Won't show any marks if SHOWN_MARKS.NONE */
   const marksToShow = shownMarks === SHOWN_MARKS.ALL || shownMarks === SHOWN_MARKS.USER
 
   if (previousAnnotations?.length > 0 && marksToShow) {
-    // Wrapping the array in an react fragment because enzyme errors otherwise 
-    // React v16 allows this, though. 
     return (
       <>
         {previousAnnotations.map((annotation) => {
-          const annotationValuesPerFrame = (multiImageCloneMarkers)
-            ? annotation.value
-            : annotation.value.filter(value => value.frame === frame)
-          
+          const annotationMarks = annotation.toSnapshot()[0].value
+          console.log(annotationMarks)
```

Comment:
This shows correctly stored `marks` while jumping between workflow steps. The issue is that PreviousMarks isn't actually filtering per `frame` yet.

Reply:
Here, you should use [`getSnapshot(annotation)`](https://mobx-state-tree.js.org/concepts/snapshots) to debug the annotation. That's how I figured out that the annotation value didn't include all of the drawn marks.

`annotation.toSnapshot()` serialises a drawing annotation so that it can be saved to Panoptes.